EventFilter/SiPixelRawToDigi/interface/GPUSimpleVector.hpp

```diff
@@ -0,0 +1,72 @@
+//  author: Felice Pantaleo, CERN, 2018
+#ifndef GPU_SIMPLEVECTOR_HPP_
+#define GPU_SIMPLEVECTOR_HPP_
+
+namespace GPU {
+template <class T> struct SimpleVector {
+  // Constructors
+  __host__ __device__ SimpleVector(unsigned int maxSize, T *m_data = nullptr)
```

Comment:
Why `nullptr` is allowed (and even default)? From the code it looks to me that any proper use of the class requires passing the pointer to user-allocated memory to the constructor.

Consider adding a comment that the ownership of `m_data` stays within the caller.